src/test/regress/expected/multi_partitioning.out

```diff
@@ -923,15 +923,324 @@ ERROR:  no partition of relation "multi_column_partitioning_1660068" found for r
 DETAIL:  Partition key of the failing row contains (c1, c2) = (20, -20).
 CONTEXT:  while executing command on localhost:57637
 -- see data is loaded to multi-column partitioned table
-SELECT * FROM multi_column_partitioning;
+SELECT * FROM multi_column_partitioning ORDER BY 1, 2;
  c1 | c2  
 ----+-----
   1 |   1
   5 |  -5
- 19 | -19
  11 | -11
+ 19 | -19
+(4 rows)
+
+--
+-- Tests for locks on partitioned tables
+--
+CREATE TABLE partitioning_locks(id int, ref_id int, time date) PARTITION BY RANGE (time);
+-- create its partitions
+CREATE TABLE partitioning_locks_2009 PARTITION OF partitioning_locks FOR VALUES FROM ('2009-01-01') TO ('2010-01-01');
+CREATE TABLE partitioning_locks_2010 PARTITION OF partitioning_locks FOR VALUES FROM ('2010-01-01') TO ('2011-01-01');
+-- distribute partitioned table
+SELECT create_distributed_table('partitioning_locks', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- test locks on router SELECT
+BEGIN;
+SELECT * FROM partitioning_locks WHERE id = 1 ORDER BY 1, 2;
+ id | ref_id | time 
+----+--------+------
+(0 rows)
+
+SELECT relation::regclass, locktype, mode FROM pg_locks WHERE relation::regclass::text LIKE 'partitioning_locks%' ORDER BY 1, 2, 3;
+        relation         | locktype |      mode       
+-------------------------+----------+-----------------
+ partitioning_locks      | relation | AccessShareLock
+ partitioning_locks_2009 | relation | AccessShareLock
+ partitioning_locks_2010 | relation | AccessShareLock
+(3 rows)
+
+COMMIT;
+-- test locks on real-time SELECT
+BEGIN;
+SELECT * FROM partitioning_locks ORDER BY 1, 2;
+ id | ref_id | time 
+----+--------+------
+(0 rows)
+
+SELECT relation::regclass, locktype, mode FROM pg_locks WHERE relation::regclass::text LIKE 'partitioning_locks%' ORDER BY 1, 2, 3;
+        relation         | locktype |      mode       
+-------------------------+----------+-----------------
+ partitioning_locks      | relation | AccessShareLock
+ partitioning_locks_2009 | relation | AccessShareLock
+ partitioning_locks_2010 | relation | AccessShareLock
+(3 rows)
+
+COMMIT;
+-- test locks on task-tracker SELECT
+SET citus.task_executor_type TO 'task-tracker';
+BEGIN;
+SELECT * FROM partitioning_locks AS pl1 JOIN partitioning_locks AS pl2 ON pl1.id = pl2.ref_id ORDER BY 1, 2;
+ id | ref_id | time | id | ref_id | time 
+----+--------+------+----+--------+------
+(0 rows)
+
+SELECT relation::regclass, locktype, mode FROM pg_locks WHERE relation::regclass::text LIKE 'partitioning_locks%' ORDER BY 1, 2, 3;
```

Comment:
I'd be a bit more comfortable if you do `AND pid = pg_backend_pid()` to the `where` clause on all such queries.